### PR TITLE
[BALDE-89] Build balde-engine on mac

### PR DESCRIPTION
**Pull request by Blinkuu**

### CODE REVIEW

File: engine/CMakeLists.txt [hunk 1]
```diff
@@ -53,12 +53,15 @@ if(TRAVIS_BUILD)
...
-	set(DEBUG_COMPILE_FLAGS -fPIC -std=c++17 -fsanitize=leak -pedantic -Wall -Wextra -Wnon-virtual-dtor -Werror -Wcast-align -Wunused -Woverloaded-virtual -Wpedantic -Wconversion -Wnull-dereference -Wdouble-promotion -Wduplicated-cond -Wlogical-op -g)
+	set(DEBUG_COMPILE_FLAGS -fPIC -std=c++17 -fsanitize=leak -pedantic -Wall -Wextra -Wnon-virtual-dtor -Werror -Wcast-align -Wunused -Woverloaded-virtual -Wpedantic -Wconversion -Wnull-dereference -Wdouble-promotion -g)
 	set(RELEASE_COMPILE_FLAGS -std=c++17 -Ofast -Werror)
 	add_definitions(-DLINUX)
 elseif(WINDOWS_BUILD)
 	add_definitions(-DWINDOWS)
 	#TODO: Add warning and optimization flags
+elseif(APPLE_BUILD)
+	set(DEBUG_COMPILE_FLAGS -fPIC -std=c++17 -pedantic -Wall -Wextra -Wnon-virtual-dtor -Werror -Wcast-align -Wunused -Woverloaded-virtual -Wpedantic -Wconversion -Wnull-dereference -Wdouble-promotion -g)
+	set(RELEASE_COMPILE_FLAGS -std=c++17 -Ofast -Werror)
```
Comment (BIGbadEL) [inline engine/CMakeLists.txt:64]: define APPLE macro

File: engine/CMakeLists.txt [hunk 2]
```diff
@@ -136,6 +139,14 @@ if(LINUX_BUILD)
...
 		set_target_properties(bald_engine PROPERTIES COMPILE_FLAGS "-Ofast -Werror")
 	endif()
+elseif(APPLE_BUILD)
+	if(DEFINE_DEBUG)
+		set_target_properties(bald_engine PROPERTIES COTIRE_CXX_PREFIX_HEADER_INIT "src/core/pch.h")
+		set_target_properties(bald_engine PROPERTIES COMPILE_FLAGS "-pedantic -Wall -Wextra -Wnon-virtual-dtor -Werror -Wcast-align -Wunused -Woverloaded-virtual -Wpedantic -Wconversion -Wnull-dereference -Wdouble-promotion -g")
+	elseif(DEFINE_RELEASE)
+		set_target_properties(bald_engine PROPERTIES COTIRE_CXX_PREFIX_HEADER_INIT "src/core/pch.h")
+		set_target_properties(bald_engine PROPERTIES COMPILE_FLAGS "-Ofast -Werror")
+	endif()
```
Comment (BIGbadEL) [inline engine/CMakeLists.txt:150]: can we make it in one if statement -> LINUX_BUILD || APPLE_BUILD ?

File: engine/src/core/debug/bald_assert.h [hunk 3]
```diff
@@ -97,7 +97,7 @@ namespace Bald::Debug {
  * @brief Instruction that emits SIGTRAP signal.
  */
 
-#ifdef LINUX
+#if defined(LINUX) || defined(__APPLE__)
```
Comment (BIGbadEL) [inline engine/src/core/debug/bald_assert.h]: we should define our macro and use it.

File: engine/src/core/utils/time/time.h [hunk 4]
```diff
@@ -79,6 +79,8 @@ namespace Bald::Utils {
         auto now_c = std::chrono::system_clock::to_time_t(now);
 #ifdef LINUX
         auto time = std::localtime(&now_c);
+#elif __APPLE__
+        auto time = std::localtime(&now_c);
```
Comment (BIGbadEL) [inline engine/src/core/utils/time/time.h]: why elif if it looks the same on LINUX?

**Review by BIGbadEL — CHANGES_REQUESTED** [review]

File: engine/CMakeLists.txt (hunk 2 above)
Comment (Blinkuu) [reply engine/CMakeLists.txt:150]: No, we use different compiler flags for each system.

**Review by BIGbadEL — APPROVED** [review]

**Pull request merged**